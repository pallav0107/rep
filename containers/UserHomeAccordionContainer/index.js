export { default as UserHomeAccordionContainer } from './UserHomeAccordionContainer';
