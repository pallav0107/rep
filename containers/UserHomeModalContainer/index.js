export { default as UserHomeModalContainer } from './UserHomeModalContainer';
