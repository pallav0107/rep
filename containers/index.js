
import LoginContainer from './LoginContainer';
import UserHomeContainer from './UserHomeContainer';
import FormMenuContainer from './FormMenuContainer';
import BackFacilitieContainer from './BackFaciliteContainer';
export {
  FormMenuContainer,
  LoginContainer,
  UserHomeContainer,
  BackFacilitieContainer
};
