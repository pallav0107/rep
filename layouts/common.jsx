import styled from 'styled-components';
import { Flex } from 'grid-styled';

export const Container = styled(Flex)`
  width: 100%;
`;

export const Backdrop = Container.extend`
  background-color: ${({ theme }) => theme.color.background};
  height: 100%;
`;
export const ProfileBackdrop = Container.extend`
  background-color: #6f4160;
  height: 100%;
`;
