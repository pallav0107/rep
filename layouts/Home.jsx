import React from 'react';
import { compose } from 'recompose';
import { withTracking, EventTypes } from '@wdpui/common-analytics';
import { withRouter } from 'react-router';
import { Container, Backdrop } from './common';
import CreateCustomerContainer from '../containers/CreateCustomerContainer';

export const Home = props => (
  <Container column flex="1 0 auto">
    <Backdrop align="center" justify="flex-start" p={40} column>
      <CreateCustomerContainer {...props} />
    </Backdrop>
  </Container>
);

const enhance = compose(
  withRouter,
  withTracking((state, { match }) => ({
    event: EventTypes.PAGE,
    params: {
      pageName: 'Home',
      path: match.path,
    },
  }))
);

export default enhance(Home);
