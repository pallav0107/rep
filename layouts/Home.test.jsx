import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Home } from './Home';
import { Container, Backdrop } from './common';
import { CustomerSearchContainer } from '../containers/CustomerSearchContainer';
import { withTheme } from '../../utils/jest/TestUtils';
import "../setupTests"

describe('Home', () => {
  describe('testing styled-components with theming', () => {
    it('should render Backdrop', () => {
      const component = withTheme(<Backdrop />);
      const wrapper = renderer.create(component).toJSON();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
