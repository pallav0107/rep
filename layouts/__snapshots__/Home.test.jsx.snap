// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home testing styled-components with theming should render Backdrop 1`] = `
.c0 {
  width: 100%;
  background-color: #f4f3f0;
  height: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  box-sizing: border-box;
}

<div
  className="c0 c1 c2"
  is={null}
/>
`;
